### PR TITLE
feat: Add completions

### PR DESCRIPTION
**Pull request by PurpleBooth**
Completions are now part of the modern binary experience. This will add
the completions for a number of shells.

### CODE REVIEW

**Comment by PurpleBooth** [comment]
Will sort out the failing tests later, got to dash out

File: docs/cli/completion.md [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+
+Completions make it a little easier to run specdown, allowing you to press tab and have a half typed command be completed. Completions can be generated in specdown by running the completion command.
+
+## Example
+
+### Bash
+We support assorted shells 
+
+```shell,script(expected_exit_code=0)
+specdown completion Bash
```
Comment (tomphp) [inline docs/cli/completion.md:11]: I'm not sure about the capitalisation of `Bash` here? The implementation suggests that possible values are lowercase (are they case insensitive?). On a personal level, I'd prefer all lowercase unless there's a good reason?

Comment (PurpleBooth) [reply docs/cli/completion.md:11]: There wasn't other than I tried an enum first and it only let me do them as capitalised them, I'll take them out

**Comment by tomphp** [comment]
This is awesome! Thanks!

**Pull request merged**